### DISCUSSION TITLE
iOS bug report: strange backstack behavior when navigating to a file

### DISCUSSION THREAD

**Discussion by keith** [community/community#4] [Mobile] [answered]
When tapping on a link such as https://github.com/llvm/llvm-project/blob/f5314d15af4f4514103ea12c74cb208538b8bef5/clang/tools/driver/driver.cpp#L79-L109 on mobile, then tap to go back in the app, instead of going to where you previously were, such as the notifications tab, you instead go up the file tree.

**Comment by rnystrom** [comment]
Ahh I see what's going on. We push a repo + directory stack when opening a universal link so that you aren't dropped into the middle of nowhere. But that same logic is leaking into in-app links (and I think it shouldn't). Does that sound about right?

**Reply by keith** [reply]
Yep!

**Comment by rnystrom** [comment, marked as answer]
This will be fixed in the next release! Thanks for reporting. If anything about the new behavior feels off, let me know.
